### PR TITLE
[release/v1.13] string matcher: add ignore_case to StringMatcher (#9868)

### PR DESCRIPTION
**Pull request by yangminzhu**
Signed-off-by: Yangmin Zhu <ymzhu@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: cherry-pick https://github.com/envoyproxy/envoy/pull/9868 to 1.13
Risk Level: Low
Testing: Unit Tests
Docs Changes: N/A
Release Notes: N/A
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by lizan** [comment]
@PiotrSikora I think we need PR like https://github.com/envoyproxy/envoy/pull/9750/files before back porting otherwise the version output will be weird.

**Comment by PiotrSikora** [comment]
@lizan good call, see #10025 #10026.

/wait

**Comment by PiotrSikora** [comment]
@mattklein123 we probably shouldn't guard `api/` changes in stable releases on @envoyproxy/api-shepherds, since those are all backports of approved changes.

File: api/envoy/type/matcher/string.proto [hunk 1]
```diff
@@ -64,6 +64,11 @@ message StringMatcher {
     // The input string must match the regular expression specified here.
     RegexMatcher safe_regex = 5 [(validate.rules).message = {required: true}];
   }
+
+  // If true, indicates the exact/prefix/suffix matching should be case insensitive. This has no
+  // effect for the safe_regex match.
+  // For example, the matcher *data* will match both input string *Data* and *data* if set to true.
+  bool ignore_case = 6;
```
Comment (jyotimahapatra) [inline api/envoy/type/matcher/string.proto]: Can we have a description in the comments about whether this field is true/false by default?

Comment (PiotrSikora) [reply api/envoy/type/matcher/string.proto]: All `bool` fields are false by default in protobuf. See: https://developers.google.com/protocol-buffers/docs/proto3#default

Comment (jyotimahapatra) [reply api/envoy/type/matcher/string.proto]: Thank you!

**Comment by PiotrSikora** [comment]
Skipping @envoyproxy/api-shepherds review (see: #10035).

**Pull request merged**